ofono/src/gprs.c

```diff
@@ -820,6 +822,185 @@ static void pri_update_mms_context_settings(struct pri_context *ctx)
 	pri_limit_mtu(settings->interface, MAX_MMS_MTU);
 }
 
+static gboolean pri_str_changed(const char *val, const char *newval)
+{
+	return newval ? (strcmp(val, newval) != 0) : !val[0];
```

Comment:
g_strcmp0 considers NULL different from an empty string, which makes it pretty much useless in our case. The first parameter is never NULL, the second one is checked prior to calling strcmp (and would have to be checked even if g_strcmp0 was used to get around its NULL vs "" logic).